internal/services/springcloud/spring_cloud_java_deployment_resource.go

```diff
@@ -218,20 +218,35 @@ func resourceSpringCloudJavaDeploymentRead(d *pluginsdk.ResourceData, meta inter
 	if resp.Sku != nil {
 		d.Set("instance_count", resp.Sku.Capacity)
 	}
-	if resp.Properties != nil && resp.Properties.DeploymentSettings != nil {
-		settings := resp.Properties.DeploymentSettings
-		d.Set("jvm_options", settings.JvmOptions)
-		d.Set("environment_variables", flattenSpringCloudDeploymentEnvironmentVariables(settings.EnvironmentVariables))
-		d.Set("runtime_version", settings.RuntimeVersion)
-		if err := d.Set("quota", flattenSpringCloudDeploymentResourceRequests(settings.ResourceRequests)); err != nil {
-			return fmt.Errorf("setting `quota`: %+v", err)
+	if resp.Properties != nil {
+		if settings := resp.Properties.DeploymentSettings; settings != nil {
+			d.Set("environment_variables", flattenSpringCloudDeploymentEnvironmentVariables(settings.EnvironmentVariables))
+			if err := d.Set("quota", flattenSpringCloudDeploymentResourceRequests(settings.ResourceRequests)); err != nil {
+				return fmt.Errorf("setting `quota`: %+v", err)
+			}
+			if !features.ThreePointOhBeta() {
```

Comment:
isn't this feature flag around the wrong way?

Reply:
Since `cpu` and `memerory_in_gb` are not in 3.0 schema, so I set them by checking `if !features.ThreePointOhBeta() {`